client/packages/coldchain/src/Equipment/ListView/AddFromScannerButton.tsx

```diff
@@ -102,26 +107,29 @@ export const AddFromScannerButtonComponent = () => {
           parsedCatalogProperties: {},
         };
         showCreateConfirmation();
-      }
+      } else info(t('error.no-asset-create-permission'))();
     }
   };
 
   const handleClick = async (e: React.MouseEvent<HTMLButtonElement>) => {
-    if (!isConnected) {
-      show(e);
-      return;
-    }
+    const permission = UserPermission.AssetQuery;
 
-    buttonRef.current?.blur();
-    if (isScanning) {
-      stopScan();
-    } else {
-      try {
-        await startScanning(handleScanResult);
-      } catch (e) {
-        error(t('error.unable-to-start-scanning', { error: e }))();
+    if (userHasPermission(permission)) {
+      if (!isConnected) {
+        show(e);
+        return;
```

Comment:
NIT, but I think it would be a nice clean up - our typical convention when we need to check something before continuing is to do an early return, like you can see is already here with the `isConnected` check.

It's a little easier to read than having to track all way to the bottom of the if block to see what would happen without the permission.

So could be:

```ts
if (!userHasPermission(permission)) {
  info(t('error.no-asset-view-permission'))();
  return;
}
... rest of fn
```

Reply:
Awesome, have done this now!